lib/rubocop/cop/performance/count.rb

```diff
@@ -27,56 +30,67 @@ class Count < Cop
+          begin_pos = if selector.is_a?(Symbol)
+                        expression.parent.loc.selector.begin_pos if expression
                       else
-                        return unless SELECTORS.include?(second_method)
-                        expression.parent.loc.selector.begin_pos
+                        _enumerable, selector, params = *expression
+
+                        if contains_selector?(expression)
+                          expression.loc.selector.begin_pos
+                        end
                       end
 
+          return unless SELECTORS.include?(selector)
+          return if params && !params.block_pass_type?
           return if node.parent && node.parent.block_type?
 
+          begin_pos ||= node.loc.expression.begin_pos
+
           range = Parser::Source::Range.new(node.loc.expression.source_buffer,
                                             begin_pos,
                                             node.loc.expression.end_pos)
 
-          add_offense(node, range,
-                      format(MSG, first_method || second_method, third_method))
+          add_offense(node, range, format(MSG, selector, counter))
         end
 
         def autocorrect(node)
-          expression, first_method, second_method, = parse(node)
+          expression, selector, = parse(node)
 
-          return if first_method == :reject || second_method == :reject
+          selector_loc = if selector.is_a?(Symbol)
+                           expression.parent.loc.selector
+                         else
+                           _enumerable, selector, = *expression
 
-          selector = if SELECTORS.include?(first_method)
-                       expression.loc.selector
-                     else
-                       expression.parent.loc.selector
-                     end
+                           if expression.loc.respond_to?(:selector)
+                             expression.loc.selector
+                           end
+                         end
+
+          return if selector == :reject
 
           lambda do |corrector|
             range = Parser::Source::Range.new(node.loc.expression.source_buffer,
                                               node.loc.dot.begin_pos,
                                               node.loc.expression.end_pos)
             corrector.remove(range)
-            corrector.replace(selector, 'count')
+            corrector.replace(selector_loc, 'count')
           end
         end
 
         private
 
         def parse(node)
-          left, third_method = *node
-          expression, second_method = *left
-          _enumerable, first_method = *expression
+          left, counter = *node
+          expression, selector, params = *left
+
+          [expression, selector, params, counter]
+        end
 
-          [expression, first_method, second_method, third_method]
+        def contains_selector?(node)
+          node.respond_to?(:loc) && node.loc.respond_to?(:selector)
```

Comment:
The use case for this came from running this cop again the RuboCop source code.

```ruby
        def check_duplicate_methods(names)
          dups = names.each_with_object({}) do |item, accum|
            accum[item[0]] ||= []
            accum[item[0]] << item[1]
          end 

          dups.each do |method, nodes|
            next if nodes.size < 2 
            lines = nodes.map { |node| node.loc.line }

            add_offense(nodes.last,
                        :keyword,
                        format(MSG, method, lines.join(', ')))
          end 
        end
```
In this code, the line `next if nodes.size < 2 ` would cause `undefined method `loc' for :nodes:Symbol`. More specifically, the node being processed was `nodes.size`.

I have simplified down the source that causes this error, and will be adding a test case that covers this